compiler/rustc_mir_transform/src/known_panics_lint.rs

```diff
@@ -896,13 +896,19 @@ impl CanConstProp {
         };
         for (local, val) in cpv.can_const_prop.iter_enumerated_mut() {
             let ty = body.local_decls[local].ty;
-            match tcx.layout_of(param_env.and(ty)) {
-                Ok(layout) if layout.size < Size::from_bytes(MAX_ALLOC_LIMIT) => {}
-                // Either the layout fails to compute, then we can't use this local anyway
-                // or the local is too large, then we don't want to.
-                _ => {
-                    *val = ConstPropMode::NoPropagation;
-                    continue;
+            if ty.is_union() {
+                // Do not const prop unions as they can
+                // ICE during layout calc
```

Comment:
If I put something like this here:

```text
// Set mode to `NoPropagation` because `KnownPanicsLint` does not actually const prop unions
```

it is a bit tautological IMHO. It will confuse more than explain anything.

However, real reason we've made this change is so that we don't call layout on a union because it can ICE, which is what the current comment talks about